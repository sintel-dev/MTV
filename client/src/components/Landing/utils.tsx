export const colorSchemes = {
  tag: [
    '#FFCD00', // investigate
    '#7CA8FF', // do not investigate
    '#A042FF', // postpone
    '#F64242', // problem
    '#F5FF00', // previously seen
    '#45F642', // normal
    '#C7C7C7', // untagged
  ],
};

// @TODO - later investigation for refactoring (fromIDToTag/fromTagToId)
export const tagSeq = [
  'Investigate',
  'Do not Investigate',
  'Postpone',
  'Problem',
  'Previously seen',
  'Normal',
  'Untagged',
];
const tagClassNames = ['investigate', 'not_investigate', 'postpone', 'problem', 'seen', 'normal', 'untagged'];

export const fromTagToClassName = (tag: string): string => {
  switch (tag) {
    case tagSeq[0]:
      return tagClassNames[0];
    case tagSeq[1]:
      return tagClassNames[1];
    case tagSeq[2]:
      return tagClassNames[2];
    case tagSeq[3]:
      return tagClassNames[3];
    case tagSeq[4]:
      return tagClassNames[4];
    case tagSeq[5]:
      return tagClassNames[5];
    default:
      return 'untagged';
  }
};

export const fromIDtoTag = (id: string): string => {
  switch (id) {
    case '1':
      return tagSeq[0];
    case '2':
      return tagSeq[1];
    case '3':
      return tagSeq[2];
    case '4':
      return tagSeq[3];
    case '5':
      return tagSeq[4];
    case '6':
      return tagSeq[5];
    default:
      return 'untagged';
  }
};

export const fromTagToID = (tag: string): string => {
  switch (tag) {
    case tagSeq[0]:
      return '1';
    case tagSeq[1]:
      return '2';
    case tagSeq[2]:
      return '3';
    case tagSeq[3]:
      return '4';
    case tagSeq[4]:
      return '5';
    case tagSeq[5]:
      return '6';
    default:
      return 'untagged';
  }
};

export const getTagColor = (tag: string): string => {
  let colorIdx: number | undefined;
  for (let i = 0; i < tagSeq.length; i += 1) {
    if (tagSeq[i] === tag) {
      colorIdx = i;
    }
  }
  if (colorIdx === undefined) {
    colorIdx = 6;
  }
  return colorSchemes.tag[colorIdx];
};
