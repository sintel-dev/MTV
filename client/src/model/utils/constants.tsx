export const API_URL = 'http://127.0.0.1:3000/api/v1/';
export const GOOGLE_CLIEN_ID = '796048243680-341g672umjji4m4okfgppm2ogfkrh8mj.apps.googleusercontent.com';
export const SESSION_TOKEN = 'AUTHORIZATION';
