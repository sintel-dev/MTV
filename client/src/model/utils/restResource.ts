import { AxiosRequestConfig, AxiosInstance } from 'axios';

export default class Resource<T, R> {
  /**
   *
   * @param server AxiosInstance
   * @param url Recourse end point
   */
  constructor(private server: AxiosInstance, private url: string) {}

  /**
   * Get all items
   * @param data body data - in most cases, this should be {}
   * @param params URL parameters
   */
  public all(data = {}, params = {}): Promise<R> {
    const promise = this.server.get<R>(this.url, { data, params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Get one item
   * @param id the item ID (name) to be retrieved
   * @param data body data â€“ in most cases, this should be {}
   * @param params URL parameters
   */
  public find(id, data = {}, params = {}): Promise<T> {
    const newUrl = `${this.url}${id}`;
    const promise = this.server.get<T>(newUrl, { data, params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Create an item.
   * @param data the item data to create
   * @param params URL parameters
   */
  public create(data = {}, params = {}): Promise<T> {
    const promise = this.server.post<T>(this.url, data, { params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Delete an item.
   * @param id the item ID (name) to delete
   * @returns Promise of the response's status
   */
  public delete(id): Promise<number> {
    const newUrl = `${this.url}${id}/`;
    const promise = this.server.delete<number>(newUrl);
    return promise
      .then((res) => res.status)
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Update an item.
   * @param id the item ID (name) to update
   * @param data the updated data of this item
   * @param params URL parameters
   */
  public update(id, data = {}, params = {}): Promise<T> {
    const newUrl = `${this.url}${id}/`;
    const promise = this.server.put<T>(newUrl, data, { params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Allow customize your request.
   * @param config request config
   * @param id item id
   */
  public request<Z>(config: AxiosRequestConfig = {}, id?: string): Promise<Z> {
    let newUrl = this.url;
    if (id !== undefined) {
      newUrl += `${id}/`;
    }
    const promise = this.server.get<Z>(newUrl, config);
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }
}
