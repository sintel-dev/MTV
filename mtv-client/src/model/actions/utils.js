const baseUrl = '/api/v1/';

export const api = {
    get(url, dataType = 'json') {
        const apiUrl = `${baseUrl}${url}`;
        const promise = fetch(apiUrl, {
            mode: 'cors',
            headers: {
                'Access-Control-Allow-Origin': '*',
            },
        });
        return promise.then(response => {
            if (response.status !== 204) {
                return dataType !== 'json' ? response.text : response.json();
            }
            return null;
        })
        .catch(err => console.error(err));
    },

    // post(url, data) {
    //     const apiUrl = `${baseUrl}${url}`;
    //     return fetch(apiUrl, {
    //         mode: 'cors',
    //         method: 'POST',
    //         headers: {
    //             'Access-Control-Allow-Origin': '*',
    //             'Content-Type': 'application/json',
    //             // 'Authorization': lscache.get('authorization')
    //         },
    //         body: JSON.stringify(data)
    //     });
    // }
};
