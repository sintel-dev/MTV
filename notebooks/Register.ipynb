{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{\n",
      "    \"message\": \"password has been sent to your email\"\n",
      "}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "\n",
    "url = 'http://localhost:3000/api/v1/users/signup/'\n",
    "myobj = {\n",
    "    'email': 'ustdongyu@gmail.com',\n",
    "    'name': 'liudongyu'\n",
    "}\n",
    "\n",
    "x = requests.post(url, json = myobj)\n",
    "\n",
    "print(x.text)\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.6.10 64-bit ('mtv': conda)",
   "language": "python",
   "name": "python361064bitmtvconda3405f5847e6d4fb4970930027f3fe839"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
