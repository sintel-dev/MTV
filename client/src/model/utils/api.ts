import axios, { AxiosRequestConfig, AxiosInstance } from 'axios';
import Cookies from 'js-cookie';
import { API_URL, SESSION_TOKEN } from './constants';
// import { SESSION_TOKEN } from '@nx-react/session';

import Resource from './restResource';
import {
  DatasetDataType,
  DatasetsResponse,
  DatarunDataType,
  DatarunsResponse,
  ExperimentDataType,
  ExperimentsResponse,
  PipelinesResponse,
  EventsResponse,
  CommentsResponse,
  DataResponse,
} from '../types/index';
import { PipelineDataType } from '../types/pipeline';
import { EventDataType } from '../types/event';
import { CommentDataType } from '../types/comment';

export class RestClient {
  private server: AxiosInstance;

  // default config for http request
  private requestConfig: AxiosRequestConfig = {
    baseURL: API_URL,
    // timeout: 2000,
  };

  public test: Resource<any, any>;

  public datasets: Resource<DatasetDataType, DatasetsResponse>;

  public dataruns: Resource<DatarunDataType, DatarunsResponse>;

  public experiments: Resource<ExperimentDataType, ExperimentsResponse>;

  public pipelines: Resource<PipelineDataType, PipelinesResponse>;

  public events: Resource<EventDataType, EventsResponse>;

  public comments: Resource<CommentDataType, CommentsResponse>;

  public data: Resource<any, DataResponse>;

  /**
   *
   * @param config AxiosRequestConfig
   */
  constructor(config: AxiosRequestConfig) {
    this.requestConfig = { ...this.requestConfig, ...config };
    this.server = axios.create(this.requestConfig);

    // add resources
    this.test = new Resource(this.server, 'test/');
    this.datasets = new Resource(this.server, 'datasets/');
    this.dataruns = new Resource(this.server, 'dataruns/');
    this.experiments = new Resource(this.server, 'experiments/');
    this.pipelines = new Resource(this.server, 'pipelines/');
    this.events = new Resource(this.server, 'events/');
    this.comments = new Resource(this.server, 'comments/');
    this.data = new Resource(this.server, 'data/');
    this.server.interceptors.request.use(this.requestInterceptor);
    this.server.interceptors.response.use(this.responseSuccessInterceptor, this.responseFailInterceptor);
  }

  public requestInterceptor(config) {
    const authHeader = Cookies.get(SESSION_TOKEN);
    if (!authHeader) {
      return config;
    }

    return {
      ...config,
      headers: {
        ...config.headers,
        Authorization: authHeader,
      },
    };
  }

  public responseSuccessInterceptor(response) {
    return response;
  }

  public responseFailInterceptor(error) {
    if (error.response.status === 401) {
      window.location.href = '/logout';
    }

    return Promise.reject(error);
  }
}

export default new RestClient({
  baseURL: 'http://127.0.0.1:3000/api/v1/',
});
