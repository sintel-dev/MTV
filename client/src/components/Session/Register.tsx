import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';
import { connect } from 'react-redux';
import { registerUserAction, googleRegisterAction } from 'src/model/actions/users';
import { registerStatus, googleRegisterStatus } from 'src/model/selectors/users';
import { FooterWrapper } from './FooterWrapper';
import { GoogleButton } from '../Common/GoogleButton';
import { isEmail, isName } from '../../model/utils/Utils';
import './Login.scss';
import Wrapper from './Wrapper';

export interface RegisterState {
  name: string;
  nameError: boolean;
  email: string;
  emailError: boolean;
}
export interface RegisterPayload {
  email: string;
  name: string;
}
export interface RegisterProps {
  registerStatus: 'loading' | 'success' | 'fail';
  googleRegisterStatus: 'success' | 'fail';
  register: (data: RegisterPayload) => void;
  googleRegister: (userData) => void;
  history: {
    push: (url: string) => void;
  };
}

class Register extends Component<RegisterProps, RegisterState> {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      email: '',
      nameError: false,
      emailError: false,
    };

    this.onNameChangeHandler = this.onNameChangeHandler.bind(this);
    this.onNameBlurHandler = this.onNameBlurHandler.bind(this);
    this.onEmailChangeHandler = this.onEmailChangeHandler.bind(this);
    this.onEmailBlurHandler = this.onEmailBlurHandler.bind(this);
    this.onUserRegister = this.onUserRegister.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (
      this.props.googleRegisterStatus !== prevProps.googleRegisterStatus &&
      this.props.googleRegisterStatus === 'success'
    ) {
      this.props.history.push('/');
      // this.props.fetchCurrentUser();
    }
  }

  onNameChangeHandler({ target }): void {
    this.setState({
      name: target.value,
      nameError: target.value.toString().length < 3 || !isName(target.value),
    });
  }

  onNameBlurHandler(): void {
    const { name, nameError } = this.state;
    if ((name.toString().length < 3 || !isName(name)) && !nameError) {
      this.setState({
        nameError: true,
      });
    }
  }

  onEmailChangeHandler({ target }): void {
    this.setState({
      email: target.value,
      emailError: !isEmail(target.value),
    });
  }

  onEmailBlurHandler(): void {
    const { emailError, email } = this.state;
    if (!isEmail(email) && !emailError) {
      this.setState({
        emailError: true,
      });
    }
  }

  onUserRegister() {
    const { name, email } = this.state;
    this.props.register({ name, email });
  }

  render() {
    if (this.props.registerStatus === 'success') {
      return (
        <Wrapper success>
          <div className="success-wrapper">
            <p>
              You will receive an e-mail shortly at <strong>{this.state.email}</strong> with your passkey.
            </p>
            <p>
              Please follow the instructions and then <Link to="/">log in</Link>.
            </p>
          </div>
        </Wrapper>
      );
    }
    const isLoading = this.props.registerStatus === 'loading';

    return (
      <Wrapper title="Register">
        <div className="field-wrapper">
          <TextField
            autoFocus
            fullWidth
            required
            id="name"
            type="text"
            label="Name"
            name="new-name"
            helperText={this.state.nameError ? 'Name is required' : null}
            disabled={isLoading}
            value={this.state.name}
            error={this.state.nameError}
            onChange={this.onNameChangeHandler}
            onBlur={this.onNameBlurHandler}
          />
        </div>
        <div className="field-wrapper">
          <TextField
            fullWidth
            required
            id="email"
            label="E-mail"
            type="email"
            name="new-email"
            helperText={this.state.emailError ? 'Invalid e-mail' : null}
            disabled={isLoading}
            value={this.state.email}
            error={this.state.emailError}
            onChange={this.onEmailChangeHandler}
            onBlur={this.onEmailBlurHandler}
          />
        </div>
        <div className="button-wrapper">
          <Button
            className="button"
            fullWidth
            variant="contained"
            color="primary"
            type="button"
            disabled={isLoading}
            onClick={this.onUserRegister}
          >
            {isLoading && <CircularProgress color="secondary" size={28} />}
            <span>Register</span>
          </Button>
        </div>
        <FooterWrapper>
          <p>
            Already have an account? <Link to="/">Log in</Link>.
          </p>
        </FooterWrapper>
        <FooterWrapper>
          <p>- or -</p>
        </FooterWrapper>
        <FooterWrapper>
          <GoogleButton onUserSelect={this.props.googleRegister} text="Sign Up with Google" />
        </FooterWrapper>
      </Wrapper>
    );
  }
}

export default connect(
  (state) => ({
    registerStatus: registerStatus(state),
    googleRegisterStatus: googleRegisterStatus(state),
  }),
  (dispatch: Function) => ({
    register: (data) => dispatch(registerUserAction(data)),
    googleRegister: (userData) => dispatch(googleRegisterAction(userData)),
  }),
)(Register);
