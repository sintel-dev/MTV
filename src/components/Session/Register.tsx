import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';
import { connect } from 'react-redux';
import { registerUserAction, googleRegisterAction } from 'src/model/actions/users';
import { registerStatus, googleRegisterStatus } from 'src/model/selectors/users';
import { RootState } from 'src/model/types';
import { isEmail, isName } from 'src/model/utils/Utils';
import { FooterWrapper } from './FooterWrapper';
import { GoogleButton } from '../Common/GoogleButton';
import Wrapper from './Wrapper';
import './Login.scss';

type RegisterState = {
  name: string;
  nameError: boolean;
  email: string;
  emailError: boolean;
};

type OwnProps = {
  history: {
    push: (url: string) => void;
  };
};

type StateProps = ReturnType<typeof mapState>;
type DispatchProps = ReturnType<typeof mapDispatch>;
type RegisterProps = StateProps & DispatchProps & OwnProps;

export class Register extends Component<RegisterProps, RegisterState> {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      email: '',
      nameError: false,
      emailError: false,
    };

    this.onNameChangeHandler = this.onNameChangeHandler.bind(this);
    this.onNameBlurHandler = this.onNameBlurHandler.bind(this);
    this.onEmailChangeHandler = this.onEmailChangeHandler.bind(this);
    this.onEmailBlurHandler = this.onEmailBlurHandler.bind(this);
    this.onUserRegister = this.onUserRegister.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (
      this.props.googleRegisterStatus !== prevProps.googleRegisterStatus &&
      this.props.googleRegisterStatus === 'success'
    ) {
      this.props.history.push('/');
      // this.props.fetchCurrentUser();
    }
  }

  onNameChangeHandler({ target }): void {
    this.setState({
      name: target.value,
      nameError: target.value.toString().length < 3 || !isName(target.value),
    });
  }

  onNameBlurHandler(): void {
    const { name, nameError } = this.state;
    if ((name.toString().length < 3 || !isName(name)) && !nameError) {
      this.setState({
        nameError: true,
      });
    }
  }

  onEmailChangeHandler({ target }): void {
    this.setState({
      email: target.value,
      emailError: !isEmail(target.value),
    });
  }

  onEmailBlurHandler(): void {
    const { emailError, email } = this.state;
    if (!isEmail(email) && !emailError) {
      this.setState({
        emailError: true,
      });
    }
  }

  onUserRegister() {
    const { name, email } = this.state;
    this.props.register({ name, email });
  }

  render() {
    if (this.props.registerStatus === 'success') {
      return (
        <Wrapper success>
          <div className="success-wrapper">
            <p>
              You will receive an e-mail shortly at <strong>{this.state.email}</strong> with your passkey.
            </p>
            <p>
              Please follow the instructions and then <Link to="/">log in</Link>.
            </p>
          </div>
        </Wrapper>
      );
    }
    const isLoading = this.props.registerStatus === 'loading';

    return (
      <Wrapper title="Register">
        <div className="field-wrapper">
          <TextField
            autoFocus
            fullWidth
            required
            id="name"
            type="text"
            label="Name"
            name="new-name"
            helperText={this.state.nameError ? 'Name is required' : null}
            disabled={isLoading}
            value={this.state.name}
            error={this.state.nameError}
            onChange={this.onNameChangeHandler}
            onBlur={this.onNameBlurHandler}
          />
        </div>
        <div className="field-wrapper">
          <TextField
            fullWidth
            required
            id="email"
            label="E-mail"
            type="email"
            name="new-email"
            helperText={this.state.emailError ? 'Invalid e-mail' : null}
            disabled={isLoading}
            value={this.state.email}
            error={this.state.emailError}
            onChange={this.onEmailChangeHandler}
            onBlur={this.onEmailBlurHandler}
          />
        </div>
        <div className="button-wrapper">
          <Button
            className="button"
            fullWidth
            variant="contained"
            color="primary"
            type="button"
            disabled={isLoading}
            onClick={this.onUserRegister}
          >
            {isLoading && <CircularProgress color="secondary" size={28} />}
            <span>Register</span>
          </Button>
        </div>
        <FooterWrapper>
          <p>
            Already have an account? <Link to="/">Log in</Link>.
          </p>
        </FooterWrapper>
        <FooterWrapper>
          <p>- or -</p>
        </FooterWrapper>
        <FooterWrapper>
          <GoogleButton onUserSelect={this.props.googleRegister} text="Sign Up with Google" />
        </FooterWrapper>
      </Wrapper>
    );
  }
}

const mapState = (state: RootState) => ({
  registerStatus: registerStatus(state),
  googleRegisterStatus: googleRegisterStatus(state),
});

const mapDispatch = (dispatch: Function) => ({
  register: (data) => dispatch(registerUserAction(data)),
  googleRegister: (userData) => dispatch(googleRegisterAction(userData)),
});

export default connect<StateProps, DispatchProps, {}, RootState>(mapState, mapDispatch)(Register);
