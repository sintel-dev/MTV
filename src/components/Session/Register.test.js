import React from 'react';
import { renderWithStore, TestWrapper } from 'src/tests/utils';
import { Register } from './Register';

describe('Testing Register component', () => {
  const RegisterComponent = renderWithStore(
    {},
    <TestWrapper>
      <Register />
    </TestWrapper>,
  );
  it('Should Render witout crashing', () => {
    expect(RegisterComponent).toMatchSnapshot();
  });
});
