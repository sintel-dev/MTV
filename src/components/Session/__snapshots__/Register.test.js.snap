// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Register component Should Render witout crashing 1`] = `
<div
  class="session-wrapper"
>
  <div
    class="session-wrapper__body"
  >
    <div
      class="session-wrapper__logo"
    >
      <svg
        height="110"
        version="1.1"
        viewBox="0 0 117 60"
        width="110"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          id="Dark-Mode"
          stroke="none"
          stroke-width="1"
        >
          <g
            fill="#000000"
            fill-rule="nonzero"
            id="Register"
            transform="translate(-151.000000, -70.000000)"
          >
            <g
              id="MIT_logo"
              transform="translate(151.000000, 70.000000)"
            >
              <g
                transform="translate(7.000000, 0.000000)"
              >
                <path
                  d="M0.18,0 L0.18,59.76 M20.7,0 L20.7,40.68 M41.22,0 L41.22,59.76 M61.74,0 L61.74,11.88 M82.62,19.08 L82.62,59.76"
                  id="Shape"
                  stroke="#A31F34"
                  stroke-width="12.6"
                />
                <line
                  id="Path"
                  stroke="#8A8B8C"
                  stroke-width="12.6"
                  x1="61.74"
                  x2="61.74"
                  y1="19.08"
                  y2="59.76"
                />
                <line
                  id="Path"
                  stroke="#A31F34"
                  stroke-width="11.88"
                  x1="76.32"
                  x2="109.44"
                  y1="5.94"
                  y2="5.94"
                />
              </g>
            </g>
          </g>
        </g>
      </svg>
    </div>
    <div
      class="session-wrapper__form "
    >
      <div
        class="session-wrapper__header"
      >
        <div
          class="session-wrapper__title"
        >
          Register
        </div>
      </div>
      <div
        class="field-wrapper"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink="false"
            for="name"
            id="name-label"
          >
            Name
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              autofocus=""
              class="MuiInputBase-input MuiInput-input"
              id="name"
              name="new-name"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="field-wrapper"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
            data-shrink="false"
            for="email"
            id="email-label"
          >
            E-mail
            <span
              aria-hidden="true"
              class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               *
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              id="email"
              name="new-email"
              required=""
              type="email"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="button-wrapper"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained button MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <span>
              Register
            </span>
          </span>
        </button>
      </div>
      <div
        class="footer-wrapper"
      >
        <p>
          Already have an account? 
          <a
            href="/"
          >
            Log in
          </a>
          .
        </p>
      </div>
      <div
        class="footer-wrapper"
      >
        <p>
          - or -
        </p>
      </div>
      <div
        class="footer-wrapper"
      >
        <div
          class="google-login-wrapper"
        >
          <button
            disabled=""
            style="background-color:#fff;display:inline-flex;align-items:center;color:rgba(0, 0, 0, .54);box-shadow:0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24);padding:0;border-radius:2px;border:1px solid transparent;font-size:14px;font-weight:500;font-family:Roboto, sans-serif;opacity:0.6"
            type="button"
          >
            <div
              style="margin-right:10px;background:#fff;padding:10px;border-radius:2px"
            >
              <svg
                height="18"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="#000"
                  fill-rule="evenodd"
                >
                  <path
                    d="M9 3.48c1.69 0 2.83.73 3.48 1.34l2.54-2.48C13.46.89 11.43 0 9 0 5.48 0 2.44 2.02.96 4.96l2.91 2.26C4.6 5.05 6.62 3.48 9 3.48z"
                    fill="#EA4335"
                  />
                  <path
                    d="M17.64 9.2c0-.74-.06-1.28-.19-1.84H9v3.34h4.96c-.1.83-.64 2.08-1.84 2.92l2.84 2.2c1.7-1.57 2.68-3.88 2.68-6.62z"
                    fill="#4285F4"
                  />
                  <path
                    d="M3.88 10.78A5.54 5.54 0 0 1 3.58 9c0-.62.11-1.22.29-1.78L.96 4.96A9.008 9.008 0 0 0 0 9c0 1.45.35 2.82.96 4.04l2.92-2.26z"
                    fill="#FBBC05"
                  />
                  <path
                    d="M9 18c2.43 0 4.47-.8 5.96-2.18l-2.84-2.2c-.76.53-1.78.9-3.12.9-2.38 0-4.4-1.57-5.12-3.74L.97 13.04C2.45 15.98 5.48 18 9 18z"
                    fill="#34A853"
                  />
                  <path
                    d="M0 0h18v18H0z"
                    fill="none"
                  />
                </g>
              </svg>
            </div>
            <span
              style="padding-right:10px;font-weight:500;padding-left:0;padding-top:10px;padding-bottom:10px"
            >
              Sign Up with Google
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="session-wrapper__splash"
  />
</div>
`;
